src/robot/libraries/Collections.py

```diff
@@ -727,6 +773,9 @@ def _yield_dict_diffs(self, keys, dict1, dict2):
             except AssertionError as err:
                 yield unic(err)
 
+    def _validate_dictionary(self, dictionary, position=1):
+        if is_string(dictionary) or is_number(dictionary):
+            raise TypeError("Expected %i. argument to be a dictionary, got '%s' instead." % (position, type_name(dictionary)))
 
```

Comment:
Same comments here than to `_validate_list` above. An extra comment to have two empty rows between classes as PEP-8 mandates.